types/src/auction.rs

```diff
@@ -324,15 +328,19 @@ pub trait Auction:
 
         // Process bids
         let mut bids_modified = false;
-        for bid in bids.values_mut() {
-            bids_modified = bid.process(era_end_timestamp_millis)
+        for (validator_public_key, bid) in bids.iter_mut() {
+            bids_modified = bid.process(era_end_timestamp_millis);
+
+            if evicted_validators.contains(validator_public_key) {
+                bids_modified = bid.deactivate()
```

Comment:
I believe it should be more like:

```suggestion
                if bid.deactivate() {
                    bids_modified = true;
                }
```
this way bids_modified would be set to true when one of the bids are deactivated. Otherwise if (lets say) last bid.deactivate would return false, all previous modifications would be lost?

Reply:
`deactivate` always returns true though